src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/bolts/IslCacheBolt.java

```diff
@@ -120,15 +131,34 @@ protected void handleInput(Tuple input) throws PipelineException {
 
             emit(ACTION_STREAM_ID.name(), input, new Values(flowId, direction, latency, maxLatency, maxLatencyTier2,
                     getCommandContext()));
+            emitLatencyStats(input, flowId, direction, latency);
         } else {
             unhandledInput(input);
         }
     }
 
+    private void emitLatencyStats(Tuple input, String flowId, FlowDirection direction, long latency) {
+        Map<String, String> tags = ImmutableMap.of(
+                "flowid", flowId,
+                "direction", direction.name().toLowerCase(),
+                "calculated", "true"
+        );
+
+        Datapoint datapoint = new Datapoint(metricFormatter.format("flow.rtt"),
+                System.currentTimeMillis(), tags, latency);
+        try {
+            List<Object> tsdbTuple = Collections.singletonList(Utils.MAPPER.writeValueAsString(datapoint));
+            emit(STATS_STREAM_ID.name(), input, tsdbTuple);
+        } catch (JsonProcessingException e) {
+            log.error("Couldn't create OpenTSDB tuple for flow {} latency stats", flowId, e);
+        }
+    }
+
     @Override
     public void declareOutputFields(OutputFieldsDeclarer declarer) {
         declarer.declareStream(ACTION_STREAM_ID.name(), new Fields(FLOW_ID_FIELD, FLOW_DIRECTION_FIELD,
                 LATENCY_FIELD, MAX_LATENCY_FIELD, MAX_LATENCY_TIER_2_FIELD, FIELD_ID_CONTEXT));
+        declarer.declareStream(STATS_STREAM_ID.name(), AbstractTopology.fieldMessage);
```

Comment:
You must use `org.openkilda.wfm.share.bolt.KafkaEncoder#FIELD_ID_PAYLOAD` as field name, because it is used to translate tuple field names into kafka record fields.